### PR TITLE
add strimzi operator to specified ns and refactor resource structure

### PR DESCRIPTION
**Pull request by damienomurchu**
## What
Update strimzi operator make target to deploy to specified ns. Also refactor makefile and folder structure of strimzi and kafka cr resources to align with other make targets

## Why
We want to install the operator into our own ns. Also our codebase should be consistent

## How
Creates operator group for strimzi operator
Refactors makefile and folder structure of strimzi and kafka cr resources to align with other make targets

Changes verified locally against a crc cluster.

### CODE REVIEW

File: install/resources/kafka/Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.PHONY: create/namespace
+create/namespace:
+	@echo "create namespace: $(KAFKA_CLUSTER_NAMESPACE)"
+	@sed 's/<NAMESPACE>/$(KAFKA_CLUSTER_NAMESPACE)/g' ./namespace.yaml | cat | oc apply -f -
```
Comment (jhellar) [inline install/resources/kafka/Makefile:4]: Would in this case be enough to use `oc new-project ...`?

File: install/resources/kafka/Makefile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+.PHONY: create/kafka
+create/kafka:
+	@echo "create kafka cluster: $(KAFKA_CLUSTER_NAMESPACE)"
+	@cat ./kafka.yaml | sed -e 's/<NAMESPACE>/$(KAFKA_CLUSTER_NAMESPACE)/g' | \
```
Comment (jhellar) [inline install/resources/kafka/Makefile:9]: There was suggestion from @david-martin to not include `namespace` metadata in yaml files and use `-n` parameter for `oc apply`. That could simplify things.

**Comment by jhellar** [comment]
Left 2 comments, otherwise LGTM. Verified against OSD.

File: install/resources/kafka/Makefile (hunk 1 above)
Comment (damienomurchu) [reply install/resources/kafka/Makefile:4]: In this case no, as oc new-project disallows any new ns with the prefix `openshift` to prevent conflicts with system ns's

**Comment by damienomurchu** [comment]
@david-martin @pb82 could you take a look and let me know if you're happy to see this get merged? 

I'm working on a follow-on PR for the `install/operator/monitoring` target that leans on the changes in this PR and would be handy to get this one in

File: install/resources/kafka/Makefile (hunk 2 above)
Comment (damienomurchu) [reply install/resources/kafka/Makefile:9]: Not a bad idea, I have a follow-on PR and will look to update the namespace metadata if I don't do it in this PR

File: install/Makefile [hunk 3]
```diff
@@ -1,8 +1,8 @@
-STRIMZI_OPERATOR_NAMESPACE ?= openshift-operators
-KAFKA_CLUSTER_NAMESPACE ?= openshift-kafka-cluster
+export STRIMZI_OPERATOR_NAMESPACE ?= openshift-kafka-operator
+export KAFKA_CLUSTER_NAMESPACE ?= openshift-kafka-cluster
```
Comment (david-martin) [inline install/Makefile:2]: I must have missed when the ns changed and we added the openshift prefix.
Out of interest, why the change?

File: install/resources/kafka/Makefile (hunk 1 above)
Comment (pb82) [reply install/resources/kafka/Makefile:4]: I also prefer namespace.yaml because we can apply it and don't need to ignore the error if it already exists.

File: install/Makefile (hunk 3 above)
Comment (damienomurchu) [reply install/Makefile:2]: @david-martin my understanding was that the `openshift` prefix was required to hook into the cluster prometheus - is that necessary?

Comment (david-martin) [reply install/Makefile:2]: @damienomurchu No, not necessary.

The cluster prometheus will only be used to federate some key metrics (e.g. kubelet_volume, kube_state).
Let's remove to avoid confusion

Comment (damienomurchu) [reply install/Makefile]: udpated now :+1:

**Comment by david-martin** [comment]
Giving this a run on local crc

**Review by david-martin — APPROVED** [review]
`make install/strimzi/operator` & `make install/kafka/cr` working locally.

```
oc get po --all-namespaces|grep kafka
kafka-cluster                                my-cluster-entity-operator-5cbc5d56db-mlkt2                  3/3     Running        0          6m19s
kafka-cluster                                my-cluster-kafka-0                                           2/2     Running        0          7m
kafka-cluster                                my-cluster-zookeeper-0                                       1/1     Running        0          8m2s
kafka-operator                               strimzi-cluster-operator-v0.19.0-6694dbfdd9-kwtjc            1/1     Running        0          13m
```
